src/coord/src/client.rs

```diff
@@ -433,49 +433,131 @@ impl SessionClient {
+                | ExecuteResponse::DroppedSchema
+                | ExecuteResponse::DroppedRole
+                | ExecuteResponse::DroppedComputeInstance
+                | ExecuteResponse::DroppedSource
+                | ExecuteResponse::DroppedIndex
+                | ExecuteResponse::DroppedSink
+                | ExecuteResponse::DroppedTable
+                | ExecuteResponse::DroppedView
+                | ExecuteResponse::DroppedType
+                | ExecuteResponse::DroppedSecret
+                | ExecuteResponse::DroppedConnector
+                | ExecuteResponse::EmptyQuery
+                | ExecuteResponse::Inserted(_)
+                | ExecuteResponse::StartedTransaction { duplicated: _ }
+                | ExecuteResponse::TransactionExited {
+                    tag: _,
+                    was_implicit: _,
+                }
+                | ExecuteResponse::Updated(_)
+                | ExecuteResponse::AlteredObject(_)
+                | ExecuteResponse::AlteredIndexLogicalCompaction
+                | ExecuteResponse::Deallocate { all: _ }
+                | ExecuteResponse::Prepare => {
+                    results.push(SimpleResult::Ok);
+                }
                 ExecuteResponse::SendingRows(rows) => {
-                    let response = rows.await;
-                    response
+                    let rows = match rows.await {
+                        PeekResponseUnary::Rows(rows) => rows,
+                        PeekResponseUnary::Error(e) => {
+                            results.push(SimpleResult::err(e.to_string()));
+                            continue;
+                        }
+                        PeekResponseUnary::Canceled => {
+                            results
+                                .push(SimpleResult::err("statement canceled due to user request"));
+                            continue;
+                        }
+                    };
+                    let mut sql_rows: Vec<Vec<serde_json::Value>> = vec![];
+                    let col_names = match desc.relation_desc {
+                        Some(desc) => desc.iter_names().map(|name| name.to_string()).collect(),
+                        None => vec![],
+                    };
+                    let mut datum_vec = mz_repr::DatumVec::new();
+                    for row in rows {
+                        let datums = datum_vec.borrow_with(&row);
+                        sql_rows.push(datums.iter().map(datum_to_json).collect());
+                    }
+                    results.push(SimpleResult::Rows {
+                        rows: sql_rows,
+                        col_names,
+                    });
+                }
+                ExecuteResponse::Fetch { .. }
+                | ExecuteResponse::SetVariable { .. }
+                | ExecuteResponse::Tailing { .. }
+                | ExecuteResponse::CopyTo { .. }
+                | ExecuteResponse::CopyFrom { .. }
```

Comment:
CopyFrom is ok because...there's never any copy data with it so we know it'll never insert rows?